src/main/java/games/strategy/engine/framework/headlessGameServer/AvailableGames.java

```diff
@@ -200,19 +118,53 @@ private static boolean addToAvailableGames(final URI uri, final Map<String, URI>
     return false;
   }
 
-  private static GameData getGameDataFromXml(final URI uri) {
-    if (uri == null) {
-      return null;
-    }
+  Set<String> getGameNames() {
+    return new HashSet<>(availableGames.keySet());
+  }
+
+
+  /**
+   * Returns the path to the file associated with the specified game.
+   *
+   * <p>
+   * The "path" is actually a URI in string form.
+   * </p>
+   *
+   * @param gameName The name of the game whose file path is to be retrieved; may be {@code null}.
+   *
+   * @return The path to the game file; or {@code null} if the game is not available.
+   */
+  String getGameFilePath(final String gameName) {
```

Comment:
this method was simply moved so that the method called by the constructor could be combined with the constructor. it was a bit spaghetti for the consturctor to be a one-liner to then jump to around line 75'ish